### PR TITLE
Add docker prereq, mention podman and rearrange sections

### LINKED ISSUE #31536: seccomp tutorial does not list Docker prerequisite
**This is a Bug Report**

**Problem:**
[Restrict a Container's Syscalls with seccomp](https://kubernetes.io/docs/tutorials/security/seccomp/) requires Docker so that you can run `kind`. The requirement is not explicit; if you had `kind` set up OK but using Podman, then you have met the prerequisites but the page does not work. 

**Proposed Solution:**
List Docker as a prerequisite. Explain that you could use a tool such as `podman` instead, but would need to adapt the instructions accordingly.

When clarifying, also consider that readers might be thinking of Docker as the container runtime that runs inside Kubernetes. In fact, `kind` typically interacts with a CRI-compatible runtime inside the cluster that it deploys for you, and _doesn't_ run Docker inside the cluster - see the banner at the top of https://kind.sigs.k8s.io/

**Page to Update:**
https://kubernetes.io/docs/tutorials/security/seccomp/

**Additional Information:**
/language en

### PR DESCRIPTION
**Pull request by mrgiles**
Add docker prerequisite comment, mention podman as an alternative and rearrange lab sections. Fixes #31536

### CODE REVIEW

**Comment by mrgiles** [comment]
@sftim, @mtardy, here's a Draft PR based on Issue 31536. I run the tutorial using both Docker and Podman. As Tim mentioned, Podman requires a couple of changes that are documented in the Kind instructions for rootless containers. I moved the first lab to the end and changed a few little things that I felt weren't clear while going through the labs myself. Thanks.

**Comment by mrgiles** [comment]
@sftim, @mtardy, @saschagrunert, I could add some instructions on how to set up the labs for Podman, but it may be out of scope in this context. Thoughts? Thanks

**Comment by mrgiles** [comment]
Can you provide some feedback on this PR @stormqueen1990? Thanks!

**Comment by mtardy** [comment]
> @sftim, @mtardy, @saschagrunert, I could add some instructions on how to set up the labs for Podman, but it may be out of scope in this context. Thoughts? Thanks

maybe that can be interesting, I don't really know the policy regarding using podman. I guess the idea is to not only propose Docker as an alternative, so maybe showing the steps with podman, if they are simple would be cool.

On another subject, if you think your PR is ready for review do not hesitate to remove it from its draft state.

**Review by stormqueen1990 — COMMENTED** [review]
Text-wise, these changes make sense to me. I unfortunately don't have enough experience to chime in on the podman part.

However, I did notice this tutorial does not work in Apple Silicon machines as-is. I built an image myself to try it out, since the `hashicorp/http-echo` repository does not have any `linux/arm64` images (the repository also seems to not be updated frequently).

**Comment by mtardy** [comment]
> Text-wise, these changes make sense to me. I unfortunately don't have enough experience to chime in on the podman part.
> 
> However, I did notice this tutorial does not work in Apple Silicon machines as-is. I built an image myself to try it out, since the `hashicorp/http-echo` repository does not have any `linux/arm64` images (the repository also seems to not be updated frequently).

if you need an image that does the same thing and run on arm64, we had a similar issue in the past https://github.com/kubernetes/website/pull/37383

**Comment by mrgiles** [comment]
Thanks for the review, @stormqueen1990 and @mtardy.

I did the rearrangement and fixes in the same PR because it's what they ask  in the issue (31536)

Regarding switching the container image from `http-echo` to `agnhost`, that would require a major rewrite of the tutorial and the accompanying lab manifests that are downloaded from `k8s.io/examples/pods/security/seccomp/profiles`.

It would almost be easier to direct the user to create their own `http-echo` aarm image (like Mauren did) than rewrite all the labs.

Note: I tried to use `agnhost` for one of the labs and I found it non-trivial to write a manifest to replace the current ones using `http-echo`. Lots of `CrashLoopBackOff` errors...

It looks like the [source code](https://github.com/hashicorp/http-echo/pull/3) update for the `http-echo` Docker file has been available from Hashicorp for a while but nobody bothered to build new images and upload them to Docker Hub.

**Comment by sftim** [comment]
If we - or CNCF - can make our own sample container image that's not `agnhost`, I'd prefer that. Needs help from a few areas though.

**Comment by mtardy** [comment]
> I did the rearrangement and fixes in the same PR because it's what they ask  in the issue (31536)


I just mentioned commit, same PR is okay

**Comment by mrgiles** [comment]
@stormqueen1990, @sftim, is the inability to run the lab on Apple Silicon using the existing `http-echo` image a showstopper? Should I close this PR? The tutorial page is still available, though, should it be taken down until the container image can be replaced with one that runs on Apple hardware? Or should we add a warning note that states this limitation? Thanks!

**Comment by stormqueen1990** [comment]
> If we - or CNCF - can make our own sample container image that's not `agnhost`, I'd prefer that. Needs help from a few areas though.

Hi there, @sftim and @mrgiles 👋

Is this something that could be expedited by opening a PR in another project in the Kubernetes org?

I have a sample Dockerfile I used to test the tutorial before that could potentially be expanded to build a multi-arch image.

File: content/en/docs/tutorials/security/seccomp.md [hunk 1]
```diff
@@ -39,6 +39,12 @@ profiles that give only the necessary privileges to your container processes.
 In order to complete all steps in this tutorial, you must install
 [kind](/docs/tasks/tools/#kind) and [kubectl](/docs/tasks/tools/#kubectl).
 
+The commands used in the tutorial assume that you are using
+[Docker](https://www.docker.com/) as your container runtime. You could also use
```
[the remark this answers is not in the document]
Comment (mrgiles) [reply content/en/docs/tutorials/security/seccomp.md]: Done. Thanks for reviewing

**Review by sftim — COMMENTED** [review]
Thanks

/lgtm

**Review by pjbgf — COMMENTED** [review]
/lgtm

**Pull request merged**